contracts/SpokePool.sol

```diff
@@ -1621,6 +1667,118 @@ abstract contract SpokePool is
+        // event to assist the Dataworker in knowing when to return funds back to the HubPool that can no longer
+        // be used for a slow fill execution.
+        bool replacedSlowFillExecution = !relayExecution.slowFill &&
+            relayFills[relayExecution.relayHash] == uint256(FillStatus.RequestedSlowFill);
+
+        // @dev This function doesn't support partial fills. Therefore, we associate the relay hash with
+        // an enum tracking its fill status. All filled relays, whether slow or fast fills, are set to the Filled
+        // status. However, we also use this slot to track whether this fill had a slow fill requested. Therefore
+        // we can include a bool in the FilledRelay event making it easy for the dataworker to compute if this
+        // fill was a fast fill that replaced a slow fill and therefore this SpokePool has excess funds that it
+        // needs to send back to the HubPool.
+        if (relayFills[relayHash] == uint256(FillStatus.Filled)) revert RelayFilled();
+        relayFills[relayHash] = uint256(FillStatus.Filled);
+
+        uint256 amountToSend = relayExecution.updatedOutputAmount;
+
+        // This can only happen in a slow fill, where the contract is funding the relay.
+        int256 payoutAdjustmentPct = relayExecution.payoutAdjustmentPct;
+        if (payoutAdjustmentPct != 0) {
+            // If payoutAdjustmentPct is positive, then the recipient will receive more than the amount they
+            // were originally expecting. If it is negative, then the recipient will receive less.
+            // -1e18 is -100%. Because we cannot pay out negative values, that is the minimum.
+            // Moreover, allow the payout adjustment to go up to 1000% (i.e. 11x).
+            // This is a sanity check to ensure the payouts do not grow too large via some sort of issue in bundle
+            // construction.
+
+            if (payoutAdjustmentPct < -1e18 || payoutAdjustmentPct > 100e18) revert InvalidPayoutAdjustmentPct();
+
+            // Note: since _computeAmountPostFees is typically intended for fees, the signage must be reversed.
+            amountToSend = _computeAmountPostFees(amountToSend, -payoutAdjustmentPct);
+        }
+
+        // If relayer and receiver are the same address, there is no need to do any transfer, as it would result in no
+        // net movement of funds.
+        // Note: this is important because it means that relayers can intentionally self-relay in a capital efficient
+        // way (no need to have funds on the destination).
+        // If this is a slow fill, we can't exit early since we still need to send funds out of this contract
+        // since there is no "relayer".
+        bool isSlowFill = relayExecution.slowFill;
+        address recipientToSend = relayExecution.updatedRecipient;
+        if (msg.sender == recipientToSend && !isSlowFill) return;
```

Comment:
Does this mean we won't emit the FilledUSSRelay in the case that it's a self-relay? That seems like an issue, no?

Reply:
Ahh yes good catch. Great catch.